### ISSUE TITLE
Unit tests for table_drop_column function

### ISSUE THREAD

**Issue by billgeo** [linz/postgresql-tableversion#106]
During testing for https://github.com/linz/linz_bde_uploader_private/issues/28 we found that either one of both of these statements didn't run properly. Not sure why, but they could be good candidates for good unit testing?

```
PERFORM @extschema@.ver_create_table_functions(p_schema_name, p_table_name, v_key_col);
PERFORM @extschema@.ver_create_version_trigger(p_schema_name, p_table_name, v_key_col); 
```

https://github.com/linz/postgresql-tableversion/blob/bd2e7e44a1da7672cc48e7e509a9db5b018e6af7/sql/16-table_drop_column.sql#L36

**Comment by palmerj** [comment]
Thanks. Actually it was the DROP COLUMN for the revision tables that didn't occur for the recent patch

**Comment by strk** [comment]
There's a test _using_ that function, but not yet checking all the effects. As the test was added by @jrlinz who was experiencing issues on some production bde processor I suspect this may be the same issue. I think @jrlinz fixed that by upgrading `table_version`.

**Comment by strk** [comment]
@palmerj is it correct to drop column from a revision table ? It would loose history (although schema changes don't seem to be considered part of history?) - current master branch of table_version does drop the column from the revision table too (I'm adding tests now)

**Comment by palmerj** [comment]
Yeah good point. Could you please review a better way to deal with this? Really we shouldn't allow deletes of columns.

**Comment by palmerj** [comment]
Note TRT columns still exists in production at the moment.

**Comment by strk** [comment]
> Note TRT columns still exists in production at the moment.

Is there a ticket specifically about this ?
Because it does look like current table_version works fine..

**Comment by strk** [comment]
> Could you please review a better way to deal with this?
> Really we shouldn't allow deletes of columns.

About about not allowing _any_ structural change ?
That is: column addition, removal, datatype change.

If a table is versioned, you cannot change it.
Rather, you create a new table.

**Comment by billgeo** [comment]
Is there some way to keep history of schema changes? Is that an option?

FYI - I raised the column deletion issues with Property Rights and explained our point of view. The last one was a tidy up before ASaTs migration, so not a great reason to do this. Still several years away. I suggested they involve us earlier in those decisions which they said they can do. If we do decide to not allow column deletions then it would be good to communicate that to them soon as possible.
